src/main/java/org/broadinstitute/dsm/pubsub/DSMtasksSubscription.java

```diff
@@ -5,31 +5,46 @@
 import com.google.cloud.pubsub.v1.AckReplyConsumer;
 import com.google.cloud.pubsub.v1.MessageReceiver;
 import com.google.cloud.pubsub.v1.Subscriber;
-import com.google.gson.Gson;
 import com.google.pubsub.v1.ProjectSubscriptionName;
 import com.google.pubsub.v1.PubsubMessage;
 import org.broadinstitute.dsm.export.ExportToES;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.util.Map;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
 
-public class ElasticExportSubscription {
+public class DSMtasksSubscription {
 
-    private static final Logger logger = LoggerFactory.getLogger(ElasticExportSubscription.class);
-    private static final Gson gson = new Gson();
+    private static final Logger logger = LoggerFactory.getLogger(DSMtasksSubscription.class);
+    public static final String TASK_TYPE = "type";
+    public static final String UPDATE_CUSTOM_WORKFLOW = "UPDATE_CUSTOM_WORKFLOW";
+    public static final String ELASTIC_EXPORT = "ELASTIC_EXPORT";
 
-    public static void subscribeElasticExport(String projectId, String subscriptionId) {
+    public static void subscribeDSMtasks(String projectId, String subscriptionId) {
         // Instantiate an asynchronous message receiver.
         MessageReceiver receiver =
                 (PubsubMessage message, AckReplyConsumer consumer) -> {
                     // Handle incoming message, then ack the received message.
                     logger.info("Got message with Id: " + message.getMessageId());
                     consumer.ack();
+                    Map<String, String> attributesMap = message.getAttributesMap();
+                    String taskType = attributesMap.get(TASK_TYPE);
                     String data = message.getData() != null ? message.getData().toStringUtf8() : null;
-                    ExportPayload payload = gson.fromJson(data, ExportPayload.class);
-                    ExportToES.exportObjectsToES(payload);
+
+                    logger.info("Task type is: " + taskType);
+
+                    switch (taskType) {
+                        case UPDATE_CUSTOM_WORKFLOW:
+                            WorkflowStatusUpdate.updateCustomWorkflow(attributesMap, data);
+                            break;
+                        case ELASTIC_EXPORT:
+                            ExportToES.exportObjectsToES(data);
+                            break;
+                        default:
+                            logger.warn("Wrong task type for a message from pubsub");
```

Comment:
It's always good practice to put break statement even if it is default to avoid possible bugs like if someone put another case below the default, since default does not have break, default content and cases' contents below default will be processed, until it finds break or end of the switch statement.